tests/webSocket-spec.coffee

```diff
@@ -0,0 +1,523 @@
+        promise.done (e) ->
+          expect(successfulMessage).toEqual e
+          ws.disconnect()
+          done()
+
+      it 'should reject the promise when error callback occurs', (done) ->
+        jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+        ws.connect()
+        message =
+          kind: 'give-me-error'
+          data:
+            cmd: 'error'
+
+        errorMessage = undefined
+        success = (data) -> 'success'
+        error = (reason) -> errorMessage = reason
+        complete = (data) -> 'complete'
+        
+        promise = ws.send(message, {success, error, complete})
+
+        promise.catch  (e) ->
+          expect(new ReallyError(errorMessage)).toEqual e
+          ws.disconnect()
+          done()
+
+
+    describe 'channel not connected', ->
+
+      it 'should buffer messages if strategy chosen is "buffer"', ->
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+        #ws.connect()
+        message = protocol.createMessage('/users')
+        
+        success = (data) -> 'success'
+        error = (reason) -> 'error'
+        complete = (data) -> 'complete'
+
+        spyOn(ws._messagesBuffer, 'push').and.callThrough()
+        promise = ws.send(message, {success, error, complete})
+        expect(ws._messagesBuffer.push).toHaveBeenCalled()#With {message, options, promise}
+        
+
+      it 'should call custom message if strategy chosen is "custom"', (done) ->
+        
+        newOptions =
+          onDisconnect: (that, messageBuffer, reallyError) ->
+            expect(that).toEqual new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', newOptions)
+            expect(messageBuffer).toBe ws._messagesBuffer
+            expect(reallyError).toEqual ReallyError
+            done()
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', newOptions)
+
+        message = protocol.createMessage('/users')
+        
+        success = (data) -> 'success'
+        error = (reason) -> 'error'
+        complete = (data) -> 'complete'
+
+        ws.send(message, {success, error, complete})
+
+      it 'should fail messages if strategy chosen is "fail" by rejecting the promise', ->
+        newOptions =
+          onDisconnect: 'fail'
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', newOptions)
+
+        message = protocol.createMessage('/users')
+        
+        success = (data) -> 'success'
+        error = (reason) -> 'error'
+        complete = (data) -> 'complete'
+
+        spyOn(ws._messagesBuffer, 'push').and.callThrough()
+        promise = ws.send(message, {success, error, complete})
+        promise.catch (e) ->
+          expect(e).toEqual new ReallyError('Connection to the server is not established')
+
+      it 'should return a promise when channel is not connected with any strategy', ->
```

Comment:
- it should be `initialized` instead of `connected`
- also I think the test should just test if it sets `onDisconnected` with default value if nothing passed and this would be sufficient